src/nwc/NWCWalletService.ts

```diff
@@ -0,0 +1,294 @@
+import {
+  nip04,
+  nip44,
+  finalizeEvent,
+  getPublicKey,
+  Event,
+  EventTemplate,
+  Relay,
+} from "nostr-tools";
+import { hexToBytes } from "@noble/hashes/utils";
+import { Subscription } from "nostr-tools/lib/types/abstract-relay";
+
+// TODO: move these imports to nwc/types
+import {
+  Nip47MakeInvoiceRequest,
+  Nip47Method,
+  Nip47NetworkError,
+  Nip47NotificationType,
+} from "./NWCClient";
+import {
+  NWCWalletServiceRequestHandler,
+  NWCWalletServiceResponse,
+  NWCWalletServiceResponsePromise,
+} from "./NWCWalletServiceRequestHandler";
+import { Nip47EncryptionType } from "./types";
+
+type NewNWCWalletServiceOptions = {
+  relayUrl: string;
+};
+
+export class NWCWalletServiceKeyPair {
+  walletSecret: string;
+  walletPubkey: string;
+  clientPubkey: string;
+  constructor(walletSecret: string, clientPubkey: string) {
+    this.walletSecret = walletSecret;
+    this.clientPubkey = clientPubkey;
+    if (!this.walletSecret) {
+      throw new Error("Missing wallet secret key");
+    }
+    if (!this.clientPubkey) {
+      throw new Error("Missing client pubkey");
+    }
+    this.walletPubkey = getPublicKey(hexToBytes(this.walletSecret));
+  }
+}
+
+export class NWCWalletService {
+  relay: Relay;
+  relayUrl: string;
+
+  constructor(options: NewNWCWalletServiceOptions) {
```

Comment:
why does the wallet service not also have the keypair?

Reply:
> do I read it correct that for each user/wallet I would need to call the subscribe function?

Yes, this is the reason. So that each connection can have unique keypairs, like we do in Alby Hub now